msg_test.go

```diff
@@ -26,6 +26,31 @@ var (
 	})
 )
 
+func TestPackNoSideEffect(t *testing.T) {
+	m := new(Msg)
+	m.SetQuestion(Fqdn("example.com."), TypeNS)
+
+	a := new(Msg)
+	o := &OPT{
+		Hdr: RR_Header{
+			Name:   ".",
+			Rrtype: TypeOPT,
+		},
+	}
+	o.SetUDPSize(DefaultMsgSize)
+
+	a.Extra = append(a.Extra, o)
+	a.SetRcode(m, RcodeBadVers)
+
+	if a.Rcode != RcodeBadVers {
+		t.Errorf("before pack: Rcode is expected to be BADVERS")
```

Comment:
This check isn’t actually test related and isn’t needed.

Reply:
I would just go with less is more and just only check after Pack()